File: src/main/java/dev/strawn/frugal/FrugalFileType.java
package dev.strawn.frugal;

import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.IconLoader;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class FrugalFileType extends LanguageFileType {
    public static final FrugalFileType INSTANCE = new FrugalFileType();
    
    private FrugalFileType() {
        super(FrugalLanguage.INSTANCE);
    }
    
    @Override
    public @NotNull String getName() {
        return "Frugal";
    }
    
    @Override
    public @NotNull String getDescription() {
        return "Frugal language files";
    }
    
    @Override
    public @NotNull String getDefaultExtension() {
        return "frugal";
    }
    
    @Override
    public @Nullable Icon getIcon() {
        return IconLoader.getIcon("/icons/frugal.svg", getClass());
    }
}

File: src/main/kotlin/dev/strawn/frugal/FrugalFileType.kt
package dev.strawn.frugal

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.util.IconLoader
import javax.swing.Icon

object FrugalFileType : LanguageFileType(FrugalLanguage) {
    
    override fun getName(): String = "Frugal"
    
    override fun getDescription(): String = "Frugal language files"
    
    override fun getDefaultExtension(): String = "frugal"
    
    override fun getIcon(): Icon? = IconLoader.getIcon("/icons/frugal.svg", javaClass)
}